common/device/api/u_device.h

```diff
@@ -428,6 +428,18 @@ int32_t uDeviceOpen(const uDeviceCfg_t *pDeviceCfg,
  */
 int32_t uDeviceClose(uDeviceHandle_t devHandle, bool powerOff);
 
+/** Attach user context to device.
+ *
+ * @param userContext A user context to set.
+ */
+void uDeviceSetUserContext(uDeviceHandle_t devHandle, void *userContext);
+
+/** Get device attached user context.
+ *
+ * @return User context that was set using @ref uDeviceSetUserContext.
+ */
+void *uDeviceGetUserContext(uDeviceHandle_t devHandle);
```

Comment:
To align with our naming conventions, `uDeviceGetUserContext()` -> `pUDeviceGetUserContext()`.